Snake moves along the sides of a rectangle.

RichardShashalevich opened the pull request.

RichardShashalevich committed:
Snake moves along the sides of a rectangle.

zayac commented on 02_snake_animation/main.go:
```diff
@@ -48,6 +48,23 @@ func moveSnake(s snake) snake {
 // Makes a single iteration for a snake.
 func step(s snake) snake {
+	x, y := s.pos.x, s.pos.y
+	if x == 50 && y == 5 {
+		s.dir.x = 0
+		s.dir.y = 1
+	}
+	if x == 50 && y == 15 {
+		s.dir.x = -1
+		s.dir.y = 0
+	}
+	if x == 5 && y == 15 {
+		s.dir.x = 0
+		s.dir.y = -1
+	}
+	if x == 5 && y == 5 {
+		s.dir.x = 1
+		s.dir.y = 0
+	}
```
Nice. To avoid many similar ifs, I'd do the following instead

```golang
for _, pivot := range []struct {
  pos, dir coord
} {
  {coord{50, 5}, coord{0, 1}},
  {coord{50, 15}, coord{-1, 0}},
  {coord{5, 15}, coord{0, -1}},
  {coord{5, 5}, coord{1, 0}},
} {
  if coord{x, y} == pivot.pos {
    s.dir = pivot.dir
  }
}
```

RichardShashalevich committed:
Added snake tail
  - Added appeles whitch increases tail

zayac commented on 04_snake_game/main.go:
```diff
@@ -49,37 +50,46 @@ type game struct {
-	return snake{coord{rand.Intn(maxX), rand.Intn(maxY)}, coord{1, 0}}
+	x, y := rand.Intn(maxX), rand.Intn(maxY)
+	return snake{
+		body: []coord{{x, y}, {x, y + 2}, {x, y}},
```
Why do the first and the last snake body parts have the same coordinate?

zayac reviewed (commented):
Can you also implement the following functionality:

* draw field borders
* implement collision with walls
* show score
* implement collisions with the snake itself?

Furthermore, can you resolve merge conflicts?

RichardShashalevich committed:
Merge branch 'main' into main

RichardShashalevich committed:
added new features
	- Added score
	- Added gameover screen
	- Added collisions

RichardShashalevich committed:
added walls with collision
- fixed bug when apple spawns outside screen

RichardShashalevich committed:
Merge branch 'prog-1:main' into main

RichardShashalevich committed:
Some bug fixes:
	- fixed bug which spawns apple in wall
	- added queue to key presses, so if your snake goes to the right, you can't get gameover, if you press up and left arrow keys at the moment of one rennder time

RichardShashalevich committed:
removed debug info

RichardShashalevich committed:
simplefied movment

RichardShashalevich commented:
There is still an issue with spawning apples, it has a chance not appear on screen

RichardShashalevich committed:
Finally fixed problem whith spawning apples

RichardShashalevich commented:
> There is still an issue with spawning apples, it has a chance not appear on screen

The problem was with spawning apple when the game starts, not when it regenerates
